aesara/tensor/elemwise.py

```diff
@@ -1364,41 +1334,42 @@ def make_node(self, input):
         from aesara.tensor.basic import as_tensor_variable
 
         input = as_tensor_variable(input)
+        inp_dims = input.type.ndim
+        inp_bdcast = input.type.broadcastable
+        inp_dtype = input.type.dtype
+        copy_op = False
 
-        if self.axis is not None:
-            for axis in self.axis:
-                if axis >= input.type.ndim or (
-                    axis < 0 and abs(axis) > input.type.ndim
-                ):
-                    raise ValueError(
-                        f"Not enough dimensions on {input} to reduce on axis {axis}"
-                    )
-        input = as_tensor_variable(input)
         axis = self.axis
         if axis is None:
-            axis = list(range(len(input.type.broadcastable)))
-        if any(a < 0 for a in axis):
-            axis2 = []
-            for a in self.axis:
+            axis = list(range(len(inp_bdcast)))
+
+        if axis is not None:
```

Comment:
Seems like this condition wouldn't be necessary at this point.  Should it be an `else`?

Reply:
It wasn't necessary, I removed it as requested.